portability-core/src/main/java/org/dataportabilityproject/cloud/google/GooglePersistentKeyValueStore.java

```diff
@@ -212,7 +249,13 @@ private Key getKey(String jobId) {
     return datastore.newKeyFactory().setKind(KIND).newKey(jobId);
   }
 
+  /**
+   * Return {@code entity}'s {@link JobState}, or null if missing.
+   *
+   * @param entity a {@link PortabilityJob}'s representation in {@link #datastore}.
+   */
   private JobState getJobState(Entity entity) {
-    return JobState.valueOf(entity.getString(PortabilityJobConverter.JOB_STATE));
+    String jobState = entity.getString(PortabilityJobConverter.JOB_STATE);
```

Comment:
Is it valid for an entity ever to have null? Is this only for the legacy, non encrypted flow case? If so, let's  add a TODO to remove.

Reply:
my understanding is it's only for the legacy non encrypted case. I'll add a TODO.